fix: Issues with scheduling timezone change without end time

nayan-bagale opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->
- Fixes Broken Ui in dark mode
- Fixes Add button always stays disabled.
- Fixes #19095 (GitHub issue number)
- Fixes CAL-5127 (Linear issue number - should be visible at the bottom of the GitHub issue description)

<!-- Please provide a loom video for visual changes to speed up reviews
 Loom Video: https://www.loom.com/
-->
Before:
[Screencast from 2025-02-05 01-53-37.webm](https://github.com/user-attachments/assets/1a2652de-8b52-4ab9-819f-5a8d44e98369)

After:
[Screencast from 2025-02-05 01-57-22.webm](https://github.com/user-attachments/assets/fa19d853-caca-4467-95ec-0eb94058477d)


## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

nayan-bagale committed:
fix: Issues with scheduling timezone change without end time

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/calcom/cal.com?pullRequest=19103) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/calcom/cal.com?pullRequest=19103) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/calcom/cal.com?pullRequest=19103) it.</sub>

nayan-bagale committed:
Merge branch 'main' into scheduling-timezone-fix

nayan-bagale commented:
Hey @CarinaWolli , I’ve changed the `DatePicker` code and added the `Calendar.tsx` component to handle single-day selection. Since `Calendar.tsx` is now being reused, should I remove the test cases for DatePicker?

nayan-bagale committed:
Merge branch 'main' into scheduling-timezone-fix

nayan-bagale committed:
test: remove DatePicker component tests

nayan-bagale commented:
Hey @CarinaWolli , I’ve changed the `DatePicker` code and added the `Calendar.tsx` component to handle single-day selection. Since `Calendar.tsx` is now being reused, So i removed the test cases for DatePicker?

nayan-bagale committed:
Merge branch 'calcom:main' into scheduling-timezone-fix

TusharBhatt1 commented on packages/ui/components/form/datepicker/DatePicker.tsx:
```diff
@@ -1,10 +1,10 @@
 
-import { Icon } from "../../icon";
+import { Button } from "../../button";
+import { Calendar } from "../date-range-picker/Calendar";
```
Try using absolute imports

nayan-bagale replied:
I did tried, but it give import error
![Screenshot from 2025-02-20 23-13-16](https://github.com/user-attachments/assets/3f6d4f58-3a9f-40f8-b5b6-bb66062500cb)

so basically we cannot use absolute imports because the components are in the same package `@calcom/ui`

TusharBhatt1 commented on packages/ui/components/form/datepicker/datepicker.test.tsx:
```diff
@@ -1,44 +0,0 @@
-/* eslint-disable playwright/missing-playwright-await */
```
Why deleted this file , instead a test as per the current implementation is recommended

TusharBhatt1 commented on packages/ui/components/form/datepicker/DatePicker.tsx:
```diff
@@ -15,20 +15,45 @@ type Props = {
+      <Popover.Root>
+        <Popover.Trigger asChild>
+          <Button
+            data-testid="date-range"
```
Ideally should be "pick-date" or similar

nayan-bagale committed:
Merge branch 'main' into scheduling-timezone-fix

TusharBhatt1 reviewed (commented):
@nayan-bagale can you add these changes exactly : https://github.com/calcom/cal.com/pull/19418/commits/9ecf7be2030fc65260902f7821398bccfa17aa44
And then let’s take this ahead

nayan-bagale committed:
Merge branch 'main' into scheduling-timezone-fix

nayan-bagale committed:
feat: implement DatePicker component with tests and update button data-testid
